### ISSUE TITLE
Photo thumbnails of private POIs can't be hidden

### ISSUE THREAD

**Issue by valleyofdawn** [IsraelHikingMap/Site#933]
## Feature
(Delete the non-relevant)

Once a picture for a private POI is taken in the field, it's thumbnail can't be hidden, only deleted.
The pictures cover the map the planned track and the recording making navigation impossible.
I suggest adding an x at the left corner of the thumbnails, letting you close them without deleting the POI.

## My environment
App on android

## Thing I think the developers should know, images, links etc.
![image](https://user-images.githubusercontent.com/689201/51632341-bcb36500-1f57-11e9-8115-77e4fb2ea6ee.png)

**Comment by HarelM** [comment]
Please define the entire UX - for example, how does one sees the image after clicking the x?
I thought about it a lot and couldn't find a good UX for this.
Ideas are welcome...

**Comment by zstadler** [comment]
- After taking the picture for a private POI and saving it, only the POI icon is visible on the map. 
- If the icon is clicked in "POI edit mode", the edit dialog will show.
- If the icon is clicked in "Read Only" or "Route edit mode", the POI details will show on the left, similar to Public POI.
   - Clicking on the picture would open it in full resolution

![image](https://user-images.githubusercontent.com/1304610/51659873-9243c500-1fb4-11e9-8b6b-789b76c6d554.png)

**Comment by HarelM** [comment]
Thanks for the idea!
The concept of private and public poi is confusing already - moving the private POI to the side bar will make it more confusing, IMO.
Regardless, the above discussion does not take into consideration the following issue: #766.
The bottom line, is that I don't have a good solution that takes into account both #766, and the visibility of the route on the map with the different states...

**Comment by valleyofdawn** [comment]
![image](https://user-images.githubusercontent.com/689201/51674131-8669fa00-1fd8-11e9-9773-b01637231ee5.png)

**Comment by HarelM** [comment]
How will it look collapsed then?

**Comment by valleyofdawn** [comment]
![image](https://user-images.githubusercontent.com/689201/51676273-d055de80-1fde-11e9-8955-0e660dff20d1.png)

**Comment by zstadler** [comment]
Here we see that a single thumbnail is problematic for navigation, having multiple thumbnails, as suggested by #766 is even worse. 

#766 may have value for embedded maps, but I fail to see its value for the user of the application.

**Comment by valleyofdawn** [comment]
Yes, maybe we should forego it altogether for the app.

**Comment by HarelM** [comment]
The title also takes space and obscures, should we remove that too?
In any case, I'm starting to get confused.
Please suggest a UX for private poi. If you think there should be a difference between iframe, mobile and desktop please write it down.
Remember that iframe are seen from mobile browsers as well (iNature is a classic example for it). So take that into consideration too.
Please keep in mind the request from my part to separate between public and private poi UX to let the user know he is dealing with different entities.
If #766 is not wanted anyone please write that too.

**Comment by valleyofdawn** [comment]
I won't have time to do this thoroughly now. But my initial insights are these:
Show thumbnails on site (mobile or PC) and in iFrames. Don't show on app. It's a cool feature for sharing and disruptive when navigating.
Use a constant size for the thumbnail rather than a constant width (portrait thumbnails are bigger now)
Use smaller thumbnails. You are not supposed to see the details, just realize something is there based on general appearance.  Also use a smaller font.
To me the font looks about right when the site is at 75% size and the landscape thumbnails look OK when it's at 50%.
Let's  forget about closing individual thumbnails for now. It's complicated and not that important.

**Comment by valleyofdawn** [comment]
The extended view is a different issue. Let's first make the thumbnails useful. Can we try the smaller thumbnails for the website/iframe and see if it makes sense?

**Comment by HarelM** [comment]
I don't know what you mean. The app and the site has the same code and I'm not planning to start a split now as I wrote above. As @zstadler mentioned and I tend to agree the thumbnail is disruptive.
Regardless, you can play around with the image size using chrome dev tools:
By clicking F12, right click on the image and select "inspect element".
On the right size you can search for width and change it from 100px to whatever suits you.
Here's a screen shot for illustration:
![image](https://user-images.githubusercontent.com/3269297/51944245-35b52f80-2424-11e9-8a9b-5103be4552ef.png)

**Comment by valleyofdawn** [comment]
I thought some pieces of code can be disabled for one of the platforms.
We are blocking several buttons on the app and iFrame, we're blocking tracking on the website, as we will also block offline maps. The same goes for thumbnails which should be disabled for the app but not the planning and sharing-oriented platforms.
If you don't want to do it, than just add the show and hide icons above, alongside the delete image icon in the extended view. I sent a mock-up pic in my second post here.
Regardless of this - POI text should be 25% smaller and thumbnail size should be 50% smaller and all thumbnails should have the same size.

**Comment by valleyofdawn** [comment]
I thing it looks nice and not too disruptive on a PC site at 67% size.
![image](https://user-images.githubusercontent.com/689201/51969660-af2d3c00-247d-11e9-83b1-da322fab478b.png)

**Comment by HarelM** [comment]
[PrivatePOI-UX.pptx](https://github.com/IsraelHikingMap/Site/files/2812516/PrivatePOI-UX.pptx)
Attached is a short presentation I did.
Here it is in words:
### On wide screens
The images will be shown when they exists - clicking on them or the text title or the marker will open the extended info dialog - which will the show the title, below the image, and below the description.
It will also have the pencil (didn't have time to do it in the presentation) to allow editing, and the upload button to convert to public POI which will open the sidebar.

### On small screens
Basically exactly the same as the above just that the image will no be displayed.

Thumbnail will be defined by a height of 64px - different orientation will cause the picture to look different, not planning on investing time in improving this except for height definition.

Not planning on changing the POI text size.

If this is not acceptable please don't start another discussion, but instead create a presentation to pitch your idea.

**Comment by zstadler** [comment]
I prefer that the user experience in all screen sizes.

Otherwise, when the route is oriented in the North-South direction, the thumbnail would cover it and deliver a worse user experience.

**Comment by zstadler** [comment]
Rotating the map in order to see the route behind one picture only to find out that a different picture now hides the route in a different place is annoying. Much more annoying than the pleasure of seeing the thumbnails.
It is not acceptable to me.

**Comment by valleyofdawn** [comment]
Following a facebook request from a user:
https://www.facebook.com/groups/994960670559126/permalink/2145875248800990/
As they stand, the POI labels cover a lot of the trail.
Things we can do to improve things:

use a smaller font
use semi-transparent background
move them behind the routes.
See example for suggestion 1 and 2

![POI](https://user-images.githubusercontent.com/689201/55399969-446bb000-5555-11e9-9678-ac5895f375f2.png)

**Comment by HarelM** [comment]
Both openlayers and Mapbox do not support moving the label behind the route.

**Comment by valleyofdawn** [comment]
Okay so we're left with smaller font and transparent background.

**Comment by zstadler** [comment]
If the POI labels cover a lot of the trail, the thumbnails are usually much bigger and cover much more of the trail and its surroundings. 

The proposal to have (semi-)transparent background for the the POI labels does not make sense while thumbnails are still shown.

**Comment by valleyofdawn** [comment]
Since both of you think the thumbnails were a bad idea, lets lose them.
I haven't heard your opinion about the hide/show button I suggested above.
Regardless of the thumbnails, the vast majority of shared routes that contain POIs have text labels and no images. We should improve their usability.
See for instance how[ the GC league](https://israelhiking.osm.org.il/share/YmZHcxnPQE) uses the shortest possible labels and posts the text outside the image to avoid concealment.

[Raz Goren](https://sites.google.com/site/mountainbikeridesinisrael/Home/mtb-israel/avital1) avoids showing them altogether, even though he needs to discuss POIs (and shows them on his other maps)

[inature ](https://inature.info/wiki/%D7%A0%D7%97%D7%9C_%D7%90%D7%9C%D7%9B%D7%A1%D7%A0%D7%93%D7%A8_-_%D7%A9%D7%91%D7%99%D7%9C_%D7%A0%D7%97%D7%9C) also often uses numbers to tackle this problem./

This is typical.

**Comment by HarelM** [comment]
I **do** think the thumbnails are needed, this is why they are still showing and this issue is still open.
I don't think we'll be able to agree without talking about it, this is why this issue is in awaiting info state.
I think that the concept of reducing the text is outdated but people are still using it although you can zoom in, zoom out and delete the POIs if they do not have value for you (this button is relatively new).
The above examples don't proof much, IMO, furthermore, I think that your suggestion about the text will still won't help those that want to keep the map as clear as possible.

Another suggestion on how to solve this would be the following:
1. On desktop - show thumbnail and text on hover and open the details dialogue on click (the dialogue part already exists today, this is how public POIs work)
2. On mobile (I don't have a good idea and I'm not sure I would like this behavior, but I'm writing this to get this thread forward) - showing nothing by default and open the details dialogue on click (the dialogue part already exists today).

In any case, no implementation is planned until we have some agreement on how to proceed.

**Comment by HarelM** [comment]
As discussed in the conference call yesterday: The current implementation is the agreed upon currently. We may visit this issue in the future.
Closing for now.

**Comment by valleyofdawn** [comment]
For future reference if we reopen this, some old design ideas on how to make the text boxes less disruptive.
![Picture1](https://user-images.githubusercontent.com/689201/57286165-61981f00-70bd-11e9-9b91-3a3497734e23.png)
